app/components/post_list/post_list.tsx

```diff
@@ -357,6 +405,28 @@ const PostList = ({
         return () => clearTimeout(t);
     }, [orderedPosts, highlightedId]);
 
+    const onScrollToEnd = () => {
+        listRef.current?.scrollToOffset({animated: true, offset: 0});
+    };
+
+    const ScrollToEndView = () => (
+        <Pressable
+            onPress={onScrollToEnd}
+            style={isNewMessagesRef.current ? styles.scrollToEndBadge : styles.scrollToEndBtn}
+        >
+            <CompassIcon
+                size={18}
+                name='arrow-down'
+                color={isNewMessagesRef.current ? theme.sidebarHeaderTextColor : theme.centerChannelColor}
+            />
+            {isNewMessagesRef.current && (
+                <Text style={styles.newMessagesText}>
+                    {location === Screens.THREAD ? 'New replies' : 'New messages'}
+                </Text>
+            )}
+        </Pressable>
+    );
```

Comment:
I would most definitely move this to it's own file

Reply:
Done 👍🏻